common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -570,6 +600,79 @@ class NetworkSensorManager : SensorManager {
         })
     }
 
+    @SuppressLint("MissingPermission")
+    private fun updateCellConnectionSensor(context: Context) {
+        if (!isEnabled(context, cellConnection) || !hasTelephony(context)) {
+            return
+        }
+
+        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+
+        val cell = telephonyManager.getAllCellInfo().findLast { cell -> cell.isRegistered } ?: return
+
+        val cellId : String = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            when (cell) {
+                is CellInfoNr -> {
+                    "nr:" + (cell.cellIdentity as CellIdentityNr).nci
+                }
+
+                is CellInfoTdscdma -> {
+                    "tdcdma:" + cell.cellIdentity.lac + ":" + cell.cellIdentity.cid
+                }
+
+                is CellInfoLte -> { //tested
+                    "lte:" + cell.cellIdentity.tac + ":" + cell.cellIdentity.ci + ":" + cell.cellIdentity.pci
+                }
+
+                is CellInfoWcdma -> { //tested
+                    "wcdma:" + cell.cellIdentity.lac + ":" + cell.cellIdentity.cid
+                }
+
+                is CellInfoCdma -> {
+                    "cdma:" + cell.cellIdentity.systemId + ":" + cell.cellIdentity.networkId + ":" + cell.cellIdentity.basestationId
+                }
+
+                is CellInfoGsm -> {
+                    "gsm:" + cell.cellIdentity.lac + ":" + ":" + cell.cellIdentity.cid
+                }
+                else -> { "" }
```

Comment:
As this variable is used for the state of the sensor it should not be blank, `STATE_UNKNOWN` might make more sense in this particular case?

Reply:
This else is actually dead code since i all the implementations of CellIdentity are covered. If there is no other proper way to do I will change it to `STATE_UNKNOWN`.

Reply:
Yea usually the `else` block is required but good to handle it incase we get some unexpected data